### ISSUE TITLE
Pre-trained weights channel ordering clarification.

### ISSUE THREAD

**Issue by norbertmarko** [tensorflow/tensorflow#42430]
Hello!

My question is about the pre-trained mobilenet-v1 model weights provided in tensorflow.keras.applications. When I use it in the TF default HWC format, everything seems to be fine, but when I set the keras backend to `'channels_first'` the weights doesn't seem to change. **Does this setting change the weight ordering, or does it only change the way the layers are interpreted?** 

I already asked it on Stackoverflow with the tensorflow TAG but no answers so far:
https://stackoverflow.com/q/63284543/10453391

If it doesn't change, what is the correct way to change the pre-trained weights ordering?` keras.utils.conv_utils.convert_kernel` is deprecated but TensorRT leans on channels first ordering heavly so I do not understand the lack of support for this.

This issue was already asked long ago but it is pretty outdated:
https://github.com/keras-team/keras/issues/3994

Thanks in advance!

**Comment by gowthamkpr** [comment]
@norbertmarko Usually in some cases changing the order of channels (with tensorflow backend) might cause some problems. Working with "channels_last" is less troublesome because of a great amount of other (non convolutional) functions that work only on the last axis.

I think this is the reason for the issue.

Can you please provide us a  minimal example(gist) like mnist, reproducing the entire issue. Thanks!
